### ISSUE TITLE
Some qol Feature requests

### ISSUE THREAD

**Issue by NikunjKhangwal** [namidaco/namida#69]
- [ ] Auto backups. I'm guessing artworks, thumbnails and video cache would be too much cpu intensive but other more necessary things like settings, playlists and history should be fine.
- [ ] Option to include artworks in re-indexing.
- [ ] Option to restore backup file from the startup screen when the app is opened for the first time after installation.
- [ ] Option to play audio only by default in youtube.
- [ ] Option to remove queues by swiping left.
- [ ] - [ ] Thumbnails animation intensity under miniplayer Customization settings change the behavior of main player as well. Both of them should be modified seperatly. For example i like to keep the Animation in main player but not in mini player.
- [ ] History shows duplicates of a single track when it's played multiple times. Instead it should show the track once with a number showing how many times it's played to make it look less cluttered. ⬇️

![photo_4_2023-11-28_19-47-09](https://github.com/namidaco/namida/assets/99175083/1eb50e9b-36ca-4f98-9655-5fcf72f39f29)

**Comment by MSOB7YY** [comment]
wow amazing suggestions, i would like to see them myself

just some notes/questions:
- auto backups will be periodically ofc (day/days/week etc) but in background or when opening namida? if background it will be real pain
- - u can already tick the "artwork" while indexing selected tracks "from advanced menu not in indexer", but good one i will add it 
  - alternatively (temprarely), u can manually clear artworks from advanced settings then do a re-index
- in which screen to swipe queue? the main queues screen?
- thumbnail animation separation: not sure about that but gonna see
- i like the history grouping, although im not sure if its really possible, will look forward to try implementing it soon

thank u so much for these awesome suggestions btw 💖

**Comment by NikunjKhangwal** [comment]
* When opening Namida works, just make sure the app automatically deletes the previous backup before creating the new one so they doesn't start to pile up.
* Yup, i'm aware of these. This option will be useful when you batch edit multiple songs using apps like [this](https://play.google.com/store/apps/details?id=com.fillobotto.mp3tagger) and they have new artworks.
* Playlists>Queues

**Comment by MSOB7YY** [comment]
auto backups are implemented with v2.0.1, the checking process happens only when u open namida

**Comment by NikunjKhangwal** [comment]
Awesome. I'll check it out.

**Comment by NikunjKhangwal** [comment]
Does the app doesn't deleted old backups automatically or mine is bugged?
![IMG_20240318_201650](https://github.com/namidaco/namida/assets/99175083/5383106d-5a32-4930-87dc-bcfaf338754a)

**Comment by NikunjKhangwal** [comment]
Also these auto backups zip files are corrupted

**Comment by MSOB7YY** [comment]
@NikunjKhangwal yep i just fixed the 0b file, its not corrupted but empty, the actual one is the one without "auto", anyways its fixed now

gonna also delete old backups, although not very sure what the max should be.. maybe 10 are good?

**Comment by MSOB7YY** [comment]
imma just make it 10. if default setting is left (auto backup each 2 days) then 20 days back is a great one i guess, the files size are not big either ways.

eta no idea 💀 i was planning for it by end of this week but things are kinda complicated, might take longer. bothways there aren't much of features, just tweaks and improvements here and there

**Comment by NikunjKhangwal** [comment]
That's fine by me as long as the feature works.

I was mainly waiting for lyrics wakelock and size gesture. No problem tho, take your time.

**Comment by MSOB7YY** [comment]
just added support for separate animations, sry it took long lmao (will be available in beta v5.5.3+ in days/weeks)

<img width="441" height="166" alt="Image" src="https://github.com/user-attachments/assets/6cdeb9ff-dd18-4c21-83f9-e4208edcfd73" />

as for repetitive history thing, it's still not really possible, i will close this issue and follow up on #746 since it has more details

thanks for the suggestions! ^^

**Comment by NikunjK8** [comment]
Thank you. Also Namida is 2 years old now? Congrats. At least I have the "before it was cool" badge.

**Comment by MSOB7YY** [comment]
hgahgahagha im just glad ur doing good, time moves fast xd
you have the "before it was cool" and also "always cool user" ^^
